pkg/k8s/apis/ciliumendpoints.cilium.io/v2/register.go

```diff
@@ -0,0 +1,138 @@
+
+	// CustomResourceDefinitionSchemaVersion is semver-conformant version of CRD schema
+	// Used to determine if CRD needs to be updated in cluster
+	CustomResourceDefinitionSchemaVersion = "1.5"
+)
+
+var (
+	// SchemeGroupVersion is group version used to register these objects
+	SchemeGroupVersion = schema.GroupVersion{
+		Group:   CustomResourceDefinitionGroup,
+		Version: CustomResourceDefinitionVersion,
+	}
+
+	// cepCRV is a minimal validation for CEP objects. Since only the agent is
+	// creating them, it is better to be permissive and have some data, if buggy,
+	// than to have no data in k8s.
+	cepCRV = apiextensionsv1beta1.CustomResourceValidation{
+		OpenAPIV3Schema: &apiextensionsv1beta1.JSONSchemaProps{},
+	}
+)
+
+// Resource takes an unqualified resource and returns a Group qualified GroupResource
+func Resource(resource string) schema.GroupResource {
+	return SchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+var (
+	// SchemeBuilder is needed by DeepCopy generator.
+	SchemeBuilder runtime.SchemeBuilder
+	// localSchemeBuilder and AddToScheme will stay in k8s.io/kubernetes.
+	localSchemeBuilder = &SchemeBuilder
+
+	// AddToScheme adds all types of this clientset into the given scheme.
+	// This allows composition of clientsets, like in:
+	//
+	//   import (
+	//     "k8s.io/client-go/kubernetes"
+	//     clientsetscheme "k8s.io/client-go/kuberentes/scheme"
+	//     aggregatorclientsetscheme "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/scheme"
+	//   )
+	//
+	//   kclientset, _ := kubernetes.NewForConfig(c)
+	//   aggregatorclientsetscheme.AddToScheme(clientsetscheme.Scheme)
+	AddToScheme = localSchemeBuilder.AddToScheme
+)
+
+func init() {
+	// We only register manually written functions here. The registration of the
+	// generated functions takes place in the generated files. The separation
+	// makes the code compile even when the generated files are missing.
+	localSchemeBuilder.Register(addKnownTypes)
+}
+
+// Adds the list of known types to api.Scheme.
+func addKnownTypes(scheme *runtime.Scheme) error {
+	scheme.AddKnownTypes(SchemeGroupVersion,
+		&CiliumEndpoint{},
+	)
+	metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
+	return nil
+}
+
+// CreateCustomResourceDefinitions creates and updates the CiliumEndpoint CRD.
+// It should be called on agent startup but is idempotent and safe to call
+// again.
+func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) error {
+	var (
+		// CustomResourceDefinitionSingularName is the singular name of custom resource definition
+		CustomResourceDefinitionSingularName = "ciliumendpoint"
+
+		// CustomResourceDefinitionPluralName is the plural name of custom resource definition
+		CustomResourceDefinitionPluralName = "ciliumendpoints"
+
+		// CustomResourceDefinitionShortNames are the abbreviated names to refer to this CRD's instances
+		CustomResourceDefinitionShortNames = []string{"cep", "ciliumep"}
```

Comment:
Did you test to upgrade a cluster with the old `cep`? I'm just curious to see if the CRD will fail to create due conflicting short names.

Reply:
Not rigourously. I will do that now. It did return objects, though, so I saw no error (which doesn't mean it worked).